source/fundamentals/crud/read-operations/retrieve.txt

```diff
@@ -0,0 +1,263 @@
+
+To find multiple documents in a collection and hold them in memory as a list, pass a query filter 
+to the ``Find()`` method that specifies the criteria of the documents you want 
+to find, then chain the ``ToList()`` or ``ToListAsync()`` method. To view a
+synchronous or asynchronous example, select the corresponding tab.
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: find-list-async
+
+      .. code-block:: csharp
+
+         var restaurants = await _restaurantsCollection.Find(filter).ToListAsync();
+
+   .. tab:: Synchronous
+      :tabid: find-list-sync
+
+      .. code-block:: csharp
+
+         var restaurants = _restaurantsCollection.Find(filter).ToList();
+
+To see a full example of using the ``Find()`` method to find multiple documents,
+see :ref:`csharp-retrieve-additional-information`.
+
+.. note:: Find All Documents
+
+   To find all documents in a collection, pass an empty filter 
+   to the ``Find()`` method.
+
+   .. code-block:: csharp
+
+      var filter = Builders<Restaurant>.Filter.Empty;
+      var allRestaurants = _restaurantsCollection.Find(filter);
+
+   To see a fully runnable example of using the ``Find()`` method to find all documents, see 
+   :ref:`csharp-retrieve-additional-information`.   
+
+Modify Find Behavior
+~~~~~~~~~~~~~~~~~~~~
+
+You can modify the behavior of the ``Find()`` method by passing
+a ``FindOptions`` object.
+
+You can configure the commonly used options with the following methods:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``BatchSize`` 
+     - | Gets or sets the number of documents to hold in a cursor at a given time.
+
+   * - ``Collation`` 
+     - | Sets the collation options. 
+
+   * - ``Comment`` 
+     - | Sets the comment to the query. To learn more about query comments, 
+         see the :manual:`$comment </reference/operator/query/comment/>` page.
+
+   * - ``Hint`` 
+     - | Sets the hint for which index to use.
+
+   * - ``MaxTime`` 
+     - | Sets the maximum execution time on the server for this operation.
+
+To see a full list of available options, see 
+`FindOptions Properties <{+api-root+}/Properties_T_MongoDB_Driver_FindOptions.htm>`__.
+
+Example
+~~~~~~~
+
+The following example performs these actions:
+
+- Finds all documents with "Pizza" in the ``cuisine`` field
+- Sets the ``BatchSize`` to ``3``
+- Stores the results in a cursor
+- Prints the number of documents currently held in the cursor
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: csharp
+
+      var filter = Builders<Restaurant>.Filter.Eq("cuisine", "Pizza");
+
+      var findOptions = new FindOptions
+      {
+         BatchSize = 3
+      };
+
+      using (var cursor = _restaurantsCollection.Find(filter, findOptions).ToCursor())
+      {
+         cursor.MoveNext();
+         WriteLine($"Number of documents in cursor: {cursor.Current.Count()}");
```

Comment:
`Console.WriteLine` here as well, and other samples to match the C# samples.

Reply:
Thanks, missed the ones in the .txt files!